swagger/swagger-generator/generator-core/src/main/java/org/apache/servicecomb/swagger/generator/core/processor/annotation/AnnotationUtils.java

```diff
@@ -127,7 +127,35 @@ public static void addResponse(Swagger swagger, ApiResponse apiResponse) {
   public static void addResponse(Swagger swagger, Operation operation, ApiResponse apiResponse) {
     ResponseConfig responseConfig = convert(apiResponse);
     generateResponse(swagger, responseConfig);
-    operation.response(responseConfig.getCode(), responseConfig.getResponse());
+    mergeResponse(operation, responseConfig);
+  }
+
+  private static void mergeResponse(Operation operation, ResponseConfig responseConfig) {
+    if (operation.getResponses() == null) {
+      operation.response(responseConfig.getCode(), responseConfig.getResponse());
+      return;
+    }
+    Response response = operation.getResponses().get(String.valueOf(responseConfig.getCode()));
+    if (response != null) {
+      Response targetResp = responseConfig.getResponse();
```

Comment:
1.targetResp is source, right?
2.logic for (response == null) is simpler, better to process first and then return, and no need write "else"